src/jit/morph.cpp

```diff
@@ -2684,7 +2685,7 @@ GenTree* Compiler::fgMakeMultiUse(GenTree** pOp)
     }
     else
     {
-        GenTree* result = fgInsertCommaFormTemp(pOp);
+        GenTree* result = fgInsertCommaFormTemp(pOp, structType);
```

Comment:
The problem was arising here due to `fgInsertCommaFormTemp` logic which requires it when pOp is structure.